packages/headless/src/utils/builder.ts

```diff
@@ -0,0 +1,43 @@
+import { computed, type ComputedRef, type HtmlHTMLAttributes, type Ref } from "vue";
+
+export type IteratedHeadlessElementFunc<T extends Record<string, unknown>> = (
+  opts: T,
+) => HtmlHTMLAttributes;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type HeadlessElementAttributes = HtmlHTMLAttributes | IteratedHeadlessElementFunc<any>;
```

Comment:
Can we use `unknown` instead of `any`?

Reply:
You restricted the generic of the `IteratedHeadlessElementFunc` with extends to be a Record, so I think it should be `Record<string, unknown>` instead of any

Reply:
This would break the generic type inference

Reply:
@larsrickert ping

Reply:
If it breaks the inference its fine like that :)